website/src/pages/api/subscriptions/[...path].ts

```diff
@@ -0,0 +1,74 @@
+import { getSession } from 'auth-astro/server';
+import { getDashboardsConfig } from '../../../config.ts';
+import type { Session } from '@auth/core/types';
+
+const API_PATHNAME_LENGTH = '/api'.length;
+
+export async function ALL({ request }: { request: Request }) {
+    const session = await getSession(request);
+
+    if (!session || !session.user?.id) {
+        return getUnauthorizedResponse(request.url);
+    }
+
+    const backendUrl = getBackendUrl(request, session);
+
+    try {
+        const response = await fetch(backendUrl, {
+            method: request.method,
+        });
+
+        return new Response(response.body, {
+            status: response.status,
+            headers: response.headers,
+        });
+    } catch (error) {
+        return getInternalErrorResponse(request.url);
+    }
+}
+
+function getBackendUrl(request: Request, session: Session) {
+    const backendEndpoint = new URL(request.url).pathname.slice(API_PATHNAME_LENGTH);
+    const backendUrl = new URL(backendEndpoint, getDashboardsConfig().backend.url);
+    backendUrl.searchParams.set('userId', session.user?.id || '');
```

Comment:
We already check in the calling function that the userId is set, so we can get rid of this confusing check:

```suggestion
function getBackendUrl(request: Request, userId: string) {
    const backendEndpoint = new URL(request.url).pathname.slice(API_PATHNAME_LENGTH);
    const backendUrl = new URL(backendEndpoint, getDashboardsConfig().backend.url);
    backendUrl.searchParams.set('userId', userId);
```